include/random_walks/shake_and_bake_walk.hpp

```diff
@@ -0,0 +1,159 @@
+        using VT = typename Polytope::VT;
+        using NT = typename Point::FT;
+
+        static constexpr NT kDefaultEpsilon = NT(1e-10);
+
+        template <typename GenericPolytope>
+        Walk(GenericPolytope&          P,
+            const Point&             boundary_pt, 
+            int                      facet_idx,    
+            RandomNumberGenerator&   rng,
+            //Mode                     m   = Mode::Original,
+            NT                       eps = kDefaultEpsilon)
+            : P_{P}, /*mode_{m}*/ epsilon_{eps}
+        {
+            initialize(boundary_pt, facet_idx, rng);
+        }
+
+        void set_epsilon(NT eps) noexcept { epsilon_ = eps; }
+        NT   get_epsilon() const noexcept { return epsilon_; }
+
+        void apply(unsigned int walk_len, RandomNumberGenerator& rng)
+        {
+            //const NT eps = epsilon_; not needed for Running
+
+            for (unsigned step = 0; step < walk_len; ++step)
+            {
+
+                Point v = GetDirection<Point>::apply(dim_, rng);
+
+                //Switching towards the inside of half-space
+                NT dot_k = A_row_k_.dot(v.getCoefficients());
+                if (dot_k > NT(0)) { v *= NT(-1); dot_k *= NT(-1); }
+
+                auto [lambda_hit, facet_new] = P_.line_positive_intersect(p_, v, Ar_, Av_);
+
+                if (!std::isfinite(lambda_hit) || lambda_hit <= NT(0) || facet_new < 0)
+                    continue;
+
+                Point y = p_ + lambda_hit * v;
+                if (!y.getCoefficients().allFinite())
+                    continue;
+
+                VT A_row_r = P_.get_facet_normal_vec(facet_new);
+                //NT dot_r   = A_row_r.dot(v.getCoefficients());
+
+                //Running
+                //if (mode_ == Mode::Running) {
+                    p_       = y;
+                    facet_idx_ = facet_new;
+                    A_row_k_   = A_row_r;
+                    Ar_.noalias() -= lambda_hit * Av_;   
+                    //continue;
+                //}
+
+                /* Original and Limping
+                NT beta;
+                if (mode_ == Mode::Original) {
+                    NT den = dot_r - dot_k;
+                    if (std::abs(den) < eps) continue;
+                    beta = std::clamp(dot_r / den, NT(0), NT(1));
+                } 
+                else {
+                    beta = -dot_k;
+                }
+
+                if (beta > NT(0) && beta <= NT(1) &&
+                    rng.sample_urdist() < beta)
+                {
+                    p_         = y;
+                    facet_idx_ = facet_new;
+                    A_row_k_   = A_row_r;
+                    Ar_.noalias() -= lambda_hit * Av_;  
+                }*/
+            }
+        }
+
+
+        const Point& getCurrentPoint() const noexcept { return p_; }
+
+    private:
+
+        void initialize(const Point& boundary_pt,
+                        int   facet_idx,
```

Comment:
I would compute the facet_idx inside this function so that the user is not forced to pass the index of the facet that the boundary point lies on.